.github/workflows/ci.yml

```
@@ -71,4 +71,40 @@ jobs:
       run: npm run build
       
     - name: Perform CodeQL Analysis
-      uses: github/codeql-action/analyze@v3
\ No newline at end of file
+      uses: github/codeql-action/analyze@v3
+
+  lineage-validation:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    needs: test
+    
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)